ethclient/ethclient.go

```diff
@@ -258,6 +258,19 @@ func (ec *Client) SubscribeNewHead(ctx context.Context, ch chan<- *types.Header)
 
 // State Access
 
+// NetworkID returns the network ID (also known as the chain ID) for this chain.
+func (ec *Client) NetworkID(ctx context.Context) (version *big.Int, err error) {
+	version = big.NewInt(0)
+
+	var ver string
+	if err = ec.c.CallContext(ctx, &ver, "net_version"); err != nil {
+		return version, err
+	}
+
+	version.SetString(ver, 10)
+	return
+}
```

Comment:
I'll remove the return value names.

Regarding the direct parsing: the result is returned from the RPC interface as a string rather than a number:

    {"jsonrpc":"2.0","id":74,"result":"1"}

so attempting to unmarshal directly in to the big.Int fails.  Is this an underlying issue with the RPC interface or should I leave the string parsing code in place?